.github/workflows/check-merge-freeze.yml

```diff
@@ -0,0 +1,22 @@
+name: Merge freeze
+
+on:
+  pull_request:
+    types: [opened, ready_for_review, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+  merge_group:
+    types: [checks_requested]
+
+jobs:
+  check-merge-freeze:
+    name: Check
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'release:prepare') }}
```

Comment:
I feel like this warrants a comment explaining how it works, at a first glance one could think that this is checking that the _current_ PR has this label, but it is actually checking IIUC if _any_ open PR has this label

Reply:
No, it is in fact checking the labels of the current PR. We wouldn't want the release PR to be frozen by itself.